### DISCUSSION TITLE
Mapping json table or tables using noderfc is it possible ????

### DISCUSSION THREAD

**Discussion by colosoderada** [SAP/node-rfc#225] [General]
Hi, I have a questions about node-rfc.

are there any examble how to mapping tables that comes from a function node-rfc to send en Resfull API ???? Or read in node information from outside (for instance in json) and pass it to node-rfc function to execute the function send the data in the tables that node-rfc function need and response in the Resfull API other tables thar SAP has responsed mapping to json, I want json to send information mapping it to the corresponding tables and response the information in the Restfull API that for instance a Android APP, Flutter or Angular has asked for. is it possible ???? Any example or help ????

If I use node-rfc passing a string and response other string is easy, but with tables from SAP to send information to SAP o response the information from one table to outside I´m getting crazy !!!!!

For instance, I have a node-rfc server that recive information from WhatsApp and send the string to SAP using a node-rfc function and push the response from SAP to WhatsApp and it is working perfect.

I´m looking for information or any example but it is impossible to find anything 

Thanks in advance.

**Comment by bsrdjan** [comment]
>  I want json to send information mapping it to the corresponding tables and response the information in the Restful API that for instance a Android APP, Flutter or Angular has asked for.

Could you share one example of required mapping, like Angular/Android RESTful API and ABAP function module to be mapped to that API?

Using `express` or similar server, Angular, React, Aurelia etc can work directly with ABAP data:
- [SAP-samples/node-rfc-samples/integration/express](https://github.com/SAP-samples/node-rfc-samples/tree/main/integration/express)
- [fundamental-tools/doc/app.md#app-server-java-node-python](https://github.com/SAP/fundamental-tools/blob/main/doc/app.md#app-server-java-node-python)

Frontend UI components bound to ABAP data structures exposed this way, can be automatically generated using [SAP Fundamental Library for ABAP](https://github.com/SAP/fundamental-tools) aka [abap-api-tools](https://www.npmjs.com/package/abap-api-tools). 

Not sure if this is exactly your use-case/scenario but might help eventually?
